src/Microsoft.ML.Tokenizers/Model/BpeOptions.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Tokenizers
+{
+    /// <summary>
+    /// Options for the BPE tokenizer.
+    /// </summary>
+    public sealed class BpeOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BpeOptions"/> class.
+        /// </summary>
+        public BpeOptions(IEnumerable<(string Token, int Id)> vocabulary)
+        {
+            if (vocabulary == null)
+            {
+                throw new ArgumentNullException(nameof(vocabulary));
+            }
+
+            Vocabulary = vocabulary;
+        }
+
+        /// <summary>
+        /// Gets or sets the vocabulary to use.
+        /// </summary>
+        public IEnumerable<(string Token, int Id)> Vocabulary { get; set; }
```

Comment:
Since we have the null check in the constructor, do we want to have the null check in the `set` here as well?

Reply:
good point. I'll remove the setter as the options has to be created with the vocabulary.